### PR TITLE
chore: DRY schedule supported actions

### PR DESCRIPTION
**Pull request by jusx**
### Changes
- Abstracted common code for Actions that support the `schedule` event.
- Added tests for the above.

### CODE REVIEW

File: lib/actions/action.js [hunk 1]
```diff
@@ -47,6 +47,18 @@ class Action extends EventAware {
     }
     this.log.info(JSON.stringify(usageLog))
   }
+
+  /**
+   * Returns the actionable issues and/or pulls if it's for a schedule event. Otherwise return the current issue or pull that is for the current event.
+   */
+  getActionables (context, results) {
+    let scheduleResults = results && results.validationSuites && results.validationSuites.length && results.validationSuites[0].schedule
```
Comment (samj1912) [inline lib/actions/action.js:55]: We could instead check for `context.event === 'schedule'` and then assign this, as it sometimes fails for other events.

Comment (shine2lay) [reply lib/actions/action.js:55]: since this piece of code is verbatim refactoring, does the issue you mentioned currently exists in our production?

**Review by shine2lay — APPROVED** [review]
LGTM! you should follow up with @samj1912 comment before merging

File: lib/actions/action.js (hunk 1 above)
Comment (samj1912) [reply lib/actions/action.js:55]: yup, although this works as expected, it's simply more explicit about what is expected to have these attributes. That way we can debug errors more easily. Currently in production it only fails if someone explicitly adds an event listener which we don't support for a particular action.

Comment (jusx) [reply lib/actions/action.js:55]: I prefer to be explicit as well. The suggestion to use `context.event` is a good one.

**Review by shine2lay — APPROVED** [review]

File: lib/actions/action.js [hunk 2]
```diff
@@ -47,6 +47,20 @@ class Action extends EventAware {
...
     this.log.info(JSON.stringify(usageLog))
   }
+
+  /**
+   * Returns the actionable issues and/or pulls if it's for a schedule event. Otherwise return the current issue or pull that is for the current event.
+   */
+  getActionables (context, results) {
+    let scheduleResults = (context.event === 'schedule')
+      ? results && results.validationSuites && results.validationSuites.length && results.validationSuites[0].schedule
+      : null
```
Comment (samj1912) [inline lib/actions/action.js:57]: Should it return an empty array instead of null here so that actionables.map doesn't fail in case nothing is present?

**Review by samj1912 — APPROVED** [review]

**Pull request merged**